imagescan/delta.go

```diff
@@ -115,13 +123,7 @@ func (d *deltaState) updateNodesUsageFromPod(v *corev1.Pod) {
 	case corev1.PodRunning, corev1.PodPending:
 		n, found := d.nodes[v.Spec.NodeName]
 		if !found {
-			n = &node{
-				name:           v.Spec.NodeName,
-				allocatableMem: &inf.Dec{},
-				allocatableCPU: &inf.Dec{},
-				pods:           make(map[types.UID]*pod),
-			}
-			d.nodes[v.Spec.NodeName] = n
+			return
```

Comment:
you will skip adding pods, but are there any guarantees, that node is removed and not in transition to the delta. I assume node addition event could race with pod addition event

Reply:
We already skip pod events for not running phase. Once node is added pod will transition to Running phase after some time. Also in kube Controller we have pods buffer https://github.com/castai/kvisor/blob/main/kube/controller.go#L208 to delay pods events